index.js

```diff
@@ -1,7 +1,78 @@
 const battleship = () => {
-  return 'The winner is...?'
+  // 1) Step 1: Create Players
+  var Player1={
+    name: " ",
+    shipCount:4,
+    gameBoard: [[0,0,0,0],[0,0,0,0],[0,0,0,0],[0,0,0,0]]
+   
 }
 
+var Player2={
+name: " ",
+shipCount:4,
+gameBoard: [[0,0,0,0],[0,0,0,0],[0,0,0,0],[0,0,0,0]]
+
+}
+Player1.name = prompt("Player 1 : Enter your Name");
+Player2.name= prompt("Player 2 : Enter your Name");
+
+//## Step 2: Randomly Add Ships to each Board
+for(let i=0;i<+(Player1.shipCount);i++){
+   var randomX=Math.floor(Math.random()*4);
+   var randomY=Math.floor(Math.random()*4);
+if(parseInt(Player1.gameBoard[randomX][randomY])!==1){
+ Player1.gameBoard[randomX][randomY]=1;}}
+
+ for(let j=0;j<+(Player2.shipCount);j++){
+  var randomX=Math.floor(Math.random()*4);
+  var randomY=Math.floor(Math.random()*4);
+if(parseInt(Player2.gameBoard[randomX][randomY])!==1){
+Player2.gameBoard[randomX][randomY]=1;}}
+ 
+console.log(Player1.gameBoard.join(", "));
+console.log(Player2.gameBoard.join(", "));
+  // Players1.gameBoard[row][col]= Math.floor(Math.random()*4);
+   //Players2.gameBoard[row][col]= Math.floor(Math.random()*4);
+ 
+var winner;
+
+ for(;;){
```

Comment:
for loop needs to have arguments inside of it, in this case you are setting up the game so you would want something like i < 0;  player.shipcount > 0; i++ or you could use a while loop for the whole game while (player.shipcount > 0)